experimental/dds/tree2/src/feature-libraries/flex-tree/lazyField.ts

```diff
@@ -388,7 +409,7 @@ export class LazySequence<TTypes extends AllowedTypes>
 		if (this.schema.types !== undefined && sourceField !== this) {
 			for (let i = sourceStart; i < sourceEnd; i++) {
 				const sourceNode = sourceField.boxedAt(sourceStart);
-				if (!this.schema.types.has(sourceNode.schema.name)) {
+				if (sourceNode === undefined || !this.schema.types.has(sourceNode.schema.name)) {
 					throw new Error("Type in source sequence is not allowed in destination.");
```

Comment:
The new condition feels like it wants a different error message, or maybe even an assert()? How do we run into this situation?

Reply:
The new condition isn't possible because of the above bounds check. This check only exists to tell the type system that.

Reply:
Prefer `...boxedAt(sourceStart) ?? fail()`. 

The affect on the code flow is the same here, but the error is wrong. If source is undefined, that doesn't indicate "Type in source sequence is not allowed in destination", it indicates "uhhh what?"